Improve code style of Send-MailMessage cmdlet

ThreeFive-O opened the pull request:
## PR Summary

This PR should improve the code readability for the cmdlet, before any other work is started. Changes:
- Change properties to auto properties
- Improve code style as suggested by CodeFactor, mainly
  - Fix comment styling
  - Insert lines after statement with curly braces

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable ~~or can only be tested interactively~~
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

ThreeFive-O committed:
Improve code style of Send-MailMessage cmdlet
Change properties to auto properties
Fix comment  styling
Insert lines after statement with curly braces

ThreeFive-O commented:
Can those suppress message attributes be removed too? e.g.
```powershell
[SuppressMessage("Microsoft.Naming", "CA1709:IdentifiersShouldBeCasedCorrectly", MessageId = "Cc")]
[SuppressMessage("Microsoft.Performance", "CA1819:PropertiesShouldNotReturnArrays")]
```
Haven't seen them in newer cmdlets. I guess some artefacts from the original code base.

iSazonov commented on src/Microsoft.PowerShell.Commands.Utility/commands/utility/Send-MailMessage.cs:
```diff
@@ -20,73 +20,41 @@ public sealed class SendMailMessage : PSCmdlet
-            }
-        }
-        private SwitchParameter _bodyashtml;
+        public SwitchParameter BodyAsHtml { get; set; } = false;
```
Is the init really needed?

ThreeFive-O replied:
Not really needed. It would be false anyway. Just make it more explicit. Same for the other inits.
I've seen it in other cmdlets and adopted this style.

iSazonov reviewed (approved):
LGTM with one minor comment.

dantraMSFT reviewed (approved):
A couple of minor comments/nits; otherwise, looks good.

ThreeFive-O committed:
Remove unnecessary intializers from properties
Fix summary comment for BodyAsHtml

ThreeFive-O commented:
@dantraMSFT and @iSazonov Thank you for the review and the help.

- Removed the initializers from the properties. For `-Encoding` paramater the initializer has to stay, because it will be null otherwise. Already present in the original code.
- Fixed the summary to be in line with the other comments.
